### PR TITLE
add CI for building dockers

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes at this moment we do not have any test all dockers are buildable and we find it out just at master when the fist build is performed

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: dockers/conda/Dockerfile [hunk 1]
```diff
@@ -23,41 +23,54 @@ RUN apt-get update && \
...
+
+COPY --chown=flash environment.yml environment.yml
 
 # conda init
 RUN conda create -y --name $CONDA_ENV python=$PYTHON_VERSION && \
     conda init bash && \
-    conda install -y python=$PYTHON_VERSION && \
+    # conda install -y python=$PYTHON_VERSION && \
+    conda install pytorch=$PYTORCH_VERSION cudatoolkit=$CUDA_VERSION --channel=pytorch && \
+    conda env update --file environment.yml && \
```
Comment (ydcjeff) [inline dockers/conda/Dockerfile:51]: why create 2 conda envs (`lightning` and `pl-env`)?

**Comment by ydcjeff** [comment]
Are we going to test this `base-cuda` build as well?
Right now, it's skipped...
https://github.com/PyTorchLightning/pytorch-lightning/runs/1087774602?check_suite_focus=true

**Comment by Borda** [comment]
> Are we going to test this `base-cuda` docker build as well?
> Right now, it's skipped... https://github.com/PyTorchLightning/pytorch-lightning/runs/1087774602?check_suite_focus=true

good catch, thx :]

File: README.md [hunk 2]
```diff
@@ -74,9 +74,9 @@ Get started with our [3 steps guide](https://pytorch-lightning.readthedocs.io/en
...
+| OSX py3.6 / py3.7 | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+testing%22) | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+testing%22) |
+| Windows py3.6 / py3.7 / py3.8 | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+testing%22) | - | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+testing%22)
```
Comment (ydcjeff) [inline README.md:79]:
```suggestion
| Linux py3.6 / py3.7 / py3.8 | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) | - | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) |
| OSX py3.6 / py3.7 | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) |
| Windows py3.6 / py3.7 / py3.8 | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) | - | - | [![CI complete testing](https://github.com/PyTorchLightning/pytorch-lightning/workflows/CI%20complete%20testing/badge.svg)](https://github.com/PyTorchLightning/pytorch-lightning/actions?query=workflow%3A%22CI+complete+testing%22) 
```

Comment (Borda) [reply README.md:79]: I do not think it is needed here, the link shows both complete and base...

**Comment by ydcjeff** [comment]
@Borda https://github.com/PyTorchLightning/pytorch-lightning/blob/a2d14306f70ec9e2a83351819d9ad4b6de9ddb26/.github/workflows/ci_pt-conda.yml#L66

this needs to change, right?

**Comment by Borda** [comment]
@tgaddair it seems we have again some problem with installing Horovoed on conda env...

**Comment by Borda** [comment]
@ydcjeff it seems there is some problem with the Cuda docker (installing Horovod) and as you already have a ready update I ll comment it now and we ll uncomment in your PR

File: .github/workflows/ci_dockers.yml [hunk 3]
```diff
@@ -52,22 +52,23 @@ jobs:
...
-
-      - name: Publish Master to Docker
-        # publish master
-        uses: docker/build-push-action@v1.1.0
-        with:
-          dockerfile: dockers/base-cuda/Dockerfile
-          build_args: PYTHON_VERSION=${{ matrix.python_version }},PYTORCH_VERSION=${{ matrix.pytorch_version }}
-          push: false
-        timeout-minutes: 40
+# TODO: uncomment this with fixing CUDA docker, no need to increase mergify count
```
Comment (nateraw) [inline .github/workflows/ci_dockers.yml:55]: why leave the comment if its in the history?

**Review by nateraw — APPROVED** [review]
looks fine 😄

File: .github/workflows/ci_dockers.yml (hunk 3 above)
Comment (Borda) [reply .github/workflows/ci_dockers.yml:55]: it is another PR #3074

**Pull request merged**